### PR TITLE
[Refactor] file_upload.rs: Use async/await instead of explicit futures

### PR DESCRIPTION
**Pull request by aliemjay**
... and rewrite `file_upload::{save_file, handle_multipart}` since they are greatly modified.
... and fix clippy warning about `create_error_response()` being always `Ok()`

### CODE REVIEW

File: src/file_upload.rs [hunk 1]
```diff
@@ -2,100 +2,72 @@ use actix_web::{
...
+                "Failed to retrieve the name of the file to upload".to_string(),
+            )
+        })?;
+
+    match std::fs::metadata(&file_path) {
+        Err(_) => Err(ContextualError::InsufficientPermissionsError(
+            file_path.display().to_string(),
         )),
-    }
+        Ok(metadata) if !metadata.is_dir() => Err(ContextualError::InvalidPathError(format!(
```
Comment (Jikstra) [inline src/file_upload.rs:60]: Never saw a match arm with an if before, that's cool!

**Review by Jikstra — APPROVED** [review]
Elegant! Very nice refactor, less code and easier to read.

**Comment by svenstaro** [comment]
I'm afraid this needs rebasing again. :)

**Review by svenstaro — APPROVED** [review]

**Comment by svenstaro** [comment]
Good stuff as usual. :)

**Pull request merged**